### PR TITLE
OCPCLOUD-2042: Update dependencies to K8s v1.27, go v1.20

### PR DESCRIPTION
**Pull request by damdo**
- bump dependencies to k8s v1.27
- fix controller-runtime breaking changes
- fix linting changes
- bump go 1.20
- introduce webhook warning changes tied to controller-runtime, left TODOs to implement them

### CODE REVIEW

File: pkg/controllers/controlplanemachinesetgenerator/controller.go [hunk 1]
```diff
@@ -250,7 +249,7 @@ func (r *ControlPlaneMachineSetGeneratorReconciler) generateControlPlaneMachineS
 
 // ensureControlPlaneMachineSet ensures that the ControlPlaneMachineSet has been created.
 func (r *ControlPlaneMachineSetGeneratorReconciler) ensureControlPlaneMachineSet(ctx context.Context, logger logr.Logger,
-	cpms *machinev1.ControlPlaneMachineSet, generatedCPMS *machinev1.ControlPlaneMachineSet) (bool, ctrl.Result, error) {
+	cpms *machinev1.ControlPlaneMachineSet, generatedCPMS *machinev1.ControlPlaneMachineSet) (bool, ctrl.Result, error) { //nolint:unparam
```
Comment (JoelSpeed) [inline pkg/controllers/controlplanemachinesetgenerator/controller.go:252]: What was the linter complaint?

File: pkg/util/watch_filters_test.go [hunk 2]
```diff
@@ -54,7 +56,7 @@ var _ = Describe("Watch Filters", func() {
 		It("returns a correct request for the cluster ControlPlaneMachineSet", func() {
 			co := configv1resourcebuilder.ClusterOperator().WithName(operatorName).Build()
 
-			Expect(clusterOperatorFilter(co)).To(ConsistOf(reconcile.Request{
+			Expect(clusterOperatorFilter(context.Background(), co)).To(ConsistOf(reconcile.Request{
```
Comment (JoelSpeed) [inline pkg/util/watch_filters_test.go:59]: Global context variable?
```suggestion
			Expect(clusterOperatorFilter(ctx, co)).To(ConsistOf(reconcile.Request{
```

File: pkg/controllers/controlplanemachineset/updates.go [hunk 3]
```diff
@@ -475,7 +475,7 @@ func (r *ControlPlaneMachineSetReconciler) createRollingUpdateReplacementMachine
 }
 
 // deleteMachine deletes the Machine provided.
-func deleteMachine(ctx context.Context, logger logr.Logger, machineProvider machineproviders.MachineProvider, outdatedMachine machineproviders.MachineInfo, namespace string) (ctrl.Result, error) {
+func deleteMachine(ctx context.Context, logger logr.Logger, machineProvider machineproviders.MachineProvider, outdatedMachine machineproviders.MachineInfo, namespace string) (ctrl.Result, error) { //nolint:unparam
```
[the remark this answers is not in the document]
Comment (damdo) [reply pkg/controllers/controlplanemachineset/updates.go:478]: ctrl.Result always being `nil`

File: pkg/controllers/controlplanemachinesetgenerator/controller.go (hunk 1 above)
Comment (damdo) [reply pkg/controllers/controlplanemachinesetgenerator/controller.go:252]: Same

File: pkg/util/watch_filters_test.go (hunk 2 above)
Comment (damdo) [reply pkg/util/watch_filters_test.go:59]: 👍

**Comment by damdo** [comment]
Flaked: 
/test unit

**Comment by EmilienM** [comment]
> /test e2e-openstack-operator

no please ignore it, I'm working on this job and the OpenStack support in CCPMSO.

**Comment by damdo** [comment]
/override e2e-openstack-operator

As it is still being developed and not testing actual merged code.

**Comment by damdo** [comment]
/override ci/prow/e2e-openstack-operator

As it is still being developed and not testing actual merged code.

**Pull request merged**